Lib/urllib/request.py

```diff
@@ -950,27 +957,51 @@ def __init__(self, password_mgr=None):
         self.passwd = password_mgr
         self.add_password = self.passwd.add_password
 
+    def _parse_realm(self, header):
+        # parse WWW-Authenticate header: accept multiple challenges per header
+        found_challenge = False
+        for mo in AbstractBasicAuthHandler.rx.finditer(header):
+            scheme, quote, realm = mo.groups()
+            if quote not in ['"',"'"]:
+                warnings.warn("Basic Auth Realm was unquoted",
+                              UserWarning, 2)
+
+            yield (scheme, realm)
+
+            found_challenge = True
+
+        if not found_challenge:
+            if header:
+                scheme = header.split()[0]
+            else:
+                scheme = ''
```

Comment:
Maybe simplify it?
```suggestion
            scheme = header.split()[0]
```

Reply:
If the header is an empty string, `"".split()[0]` raises an IndexError.